Oracle.md

```diff
@@ -0,0 +1,190 @@
+# Oracle specifications
+
+## Introduction
+
+For the purpose of these specifications, an event is a digital representation of a real world fact.
+An [oracle](./Introduction.md#Oracle) is an entity that commits to publishing a signature over a (number of) event outcome(s) ahead of time by releasing one or more [R-values](./Introduction.md#R-value) as well as necessary information for two parties to build a set of [CETs](./Introduction.md#Contract-Execution-Transaction-(CET)).
+This necessary information is committed in a so-called [_event descriptor_](#Event-descriptor) that will be further detailed in this document.
+Descriptors are classified and discovered through a namespace presently represented as Unique Resource Identifier (URI) paths.
+
+## Table of Contents
+
+- [Event descriptor](#event-descriptor)
+    - [Simple enumeration](#simple-enumeration)
+        - [Example: Weather tomorrow](#example-weather-tomorrow)
+    - [Range](#range)
+        - [Example: tomorrow's temperature](#example-tomorrows-temperature)
+    - [Large range](#large-range)
+        - [Example: BTC/USD rate](#example-btcusd-rate)
+    - [Serialization of event descriptors](#serialization-of-event-descriptors)
+        - [Version 0 `external_event_descriptor`](#version-0-external_event_descriptor)
+        - [Version 0 `enum_event_descriptor`](#version-0-enum_event_descriptor)
+        - [Version 0 `range_event_descriptor`](#version-0-range_event_descriptor)
+        - [Version 0 `large_range_event_descriptor`](#version-0-large_range_event_descriptor)
+    - [Oracle events](#oracle-events)
+        - [Version 0 `oracle_event`](#version-0-oracle_event)
+- [Oracle announcements](#oracle-announcements)
+        - [Version 0 `oracle_announcement`](#version-0-oracle_announcement)
+
+
+## Event descriptor
+
+An event descriptor provides information to clients about an event for which an oracle plans on releasing a signature over its outcome.
+The provided information should be sufficient for a client to create a set of CETs that will cover all possible outcomes of the event.
+
+Here we assume that an event outcome can be represented either as a set of strings or numbers.
+Four kinds of event outcomes are defined (note that a single event can be composed of several types).
```

Comment:
I like this idea of "Event descriptor". Are they currently stateful enough to let a client undertaking a storage failure rebuilding from scratch a CET state, if you combine them with output script descriptor ? In case of failure, you should be able to find which oracle you must watch.

cc @Christewart do we have an issue tracking all the data set required to achieve a) safe backup b) encodable in nice format for cross-client export ?

Reply:
Not sure I understand your question, but the point is that they should contain all the oracle information that you need to build CETs. I think what you would miss in case of storage failure is the information that you got from your counter-party (adaptors sigs and funding input info).

Reply:
+1 You should be able to recover all state from just the Offer, Accept, and Sign messages (where offer contains the oracle stuff like the descriptor)

Reply:
Open an issue to add fault-tolerance recommendation, we can address this separately.